airbyte-server/src/main/java/io/airbyte/server/ConnectorDefinitionSpecBackfiller.java

```diff
@@ -0,0 +1,263 @@
+              + "cannot be retrieved to be deleted, as well as their associated connections/syncs.",
+          failedBackfillImages.toString()));
+    }
+  }
+
+  public static void deleteSourceDefinition(final ConfigPersistence configPersistence, final UUID sourceDefinitionId)
+      throws JsonValidationException, ConfigNotFoundException, IOException {
+    deleteConnectorDefinition(
+        configPersistence,
+        ConfigSchema.STANDARD_SOURCE_DEFINITION,
+        ConfigSchema.SOURCE_CONNECTION,
+        SourceConnection.class,
+        SourceConnection::getSourceId,
+        SourceConnection::getSourceDefinitionId,
+        sourceDefinitionId);
+  }
+
+  public static void deleteDestinationDefinition(final ConfigPersistence configPersistence, final UUID destinationDefinitionId)
+      throws JsonValidationException, ConfigNotFoundException, IOException {
+    deleteConnectorDefinition(
+        configPersistence,
+        ConfigSchema.STANDARD_DESTINATION_DEFINITION,
+        ConfigSchema.DESTINATION_CONNECTION,
+        DestinationConnection.class,
+        DestinationConnection::getDestinationId,
+        DestinationConnection::getDestinationDefinitionId,
+        destinationDefinitionId);
+  }
+
+  private static <T> void deleteConnectorDefinition(
+                                                    final ConfigPersistence configPersistence,
+                                                    final ConfigSchema definitionType,
+                                                    final ConfigSchema connectorType,
+                                                    final Class<T> connectorClass,
+                                                    final Function<T, UUID> connectorIdGetter,
+                                                    final Function<T, UUID> connectorDefinitionIdGetter,
+                                                    final UUID definitionId)
+      throws JsonValidationException, IOException, ConfigNotFoundException {
+    final Set<T> connectors = configPersistence.listConfigs(connectorType, connectorClass)
+        .stream()
+        .filter(connector -> connectorDefinitionIdGetter.apply(connector).equals(definitionId))
+        .collect(Collectors.toSet());
+    for (final T connector : connectors) {
+      final Set<StandardSync> syncs = configPersistence.listConfigs(ConfigSchema.STANDARD_SYNC, StandardSync.class)
+          .stream()
+          .filter(sync -> sync.getSourceId().equals(connectorIdGetter.apply(connector))
+              || sync.getDestinationId().equals(connectorIdGetter.apply(connector)))
+          .collect(Collectors.toSet());
+
+      for (final StandardSync sync : syncs) {
+        configPersistence.deleteConfig(ConfigSchema.STANDARD_SYNC, sync.getConnectionId().toString());
+      }
+      configPersistence.deleteConfig(connectorType, connectorIdGetter.apply(connector).toString());
+    }
+    configPersistence.deleteConfig(definitionType, definitionId.toString());
+  }
```

Comment:
what do you think about moving this code into `ConfigRepository` so it can be shared in the future?

Reply:
I like that idea! I could see this being useful for other things in the future. Will move